src/pattern-library/examples/lazy-basic.tsx

```diff
@@ -0,0 +1,43 @@
+import { Button, lazy } from '../../';
+
+type ProfileProps = {
+  name: string;
+  description: string;
+};
+
+function Profile({ name, description }: ProfileProps) {
+  return (
+    <div className="p-4 border border-green-500 rounded">
+      <h3 className="text-green-700 font-bold">{name}</h3>
+      <p>{description}</p>
+    </div>
+  );
+}
+
+let resolve: (x: typeof Profile) => void;
+const promise = new Promise<typeof Profile>(resolve_ => (resolve = resolve_));
```

Comment:
Considering the limited scope of the pattern library (mainly used by us during development), perhaps you can use `Promise.withResolvers` here?

Reply:
The immediate reason I didn't is because the TypeScript library target doesn't make the method available (it requires es2024). In order for the pattern library to use newer APIs than are allowed in the library itself, we'll need to adjust the TS configuration. Currently the npm library and pattern library use the same TS config.